amqp-impl/src/main/java/io/streamnative/pulsar/handlers/amqp/QueueContainer.java

```diff
@@ -40,7 +60,54 @@ public static void putQueue(NamespaceName namespaceName, String queueName, AmqpQ
         });
     }
 
-    public static AmqpQueue getQueue(NamespaceName namespaceName, String queueName) {
+    public CompletableFuture<AmqpQueue> asyncGetQueue(PulsarService pulsarService, NamespaceName namespaceName,
```

Comment:
Since we have the `AmqpPulsarService` as an instance level variable, do we need to pass in the `PulsarService`?  Can we get it from the `AmqpPulsarService.getPulsarService()`?

Reply:
Yes.